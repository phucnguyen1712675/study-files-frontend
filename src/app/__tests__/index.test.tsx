import * as React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';

import { App } from '../index';

const renderer = createRenderer();

jest.mock('react-i18next', () => ({
  // this mock makes sure any components using the translate hook can use it without a warning being shown
  useTranslation: () => {
    return {
      t: (str: any) => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    };
  },
}));

describe('<App />', () => {
  it('should render and match the snapshot', () => {
    renderer.render(<App />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
