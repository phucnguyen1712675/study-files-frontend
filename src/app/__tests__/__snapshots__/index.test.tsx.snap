// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render and match the snapshot 1`] = `
<BrowserRouter>
  <Helmet
    defaultTitle="Study-files"
    defer={true}
    encodeSpecialCharacters={true}
    htmlAttributes={
      Object {
        "lang": undefined,
      }
    }
    titleTemplate="%s - Study-files"
  >
    <meta
      content="Study-files application"
      name="description"
    />
  </Helmet>
  <Context.Provider
    value={
      Object {
        "dispatch": [Function],
        "store": Object {
          "bestSellerCourses": Array [],
          "bestSellerSubCategories": Array [],
          "categories": Array [],
          "latestCourses": Array [],
          "loading": true,
          "myCourses": Array [],
          "subCategories": Array [],
          "user": Object {},
          "userId": "",
          "watchList": Array [],
        },
      }
    }
  >
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/search"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/category/:category/:subCategory"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/course/:name"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/studyPage/:name/"
      />
      <Route
        component={[Function]}
        path="/login"
      />
      <Route
        component={[Function]}
        path="/register"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/verifyEmail"
      />
      <PrivateRouteAdmin
        exact={true}
        path="/admin/users"
      >
        <Unknown />
      </PrivateRouteAdmin>
      <PrivateRouteAdmin
        exact={true}
        path="/admin/courses"
      >
        <Unknown />
      </PrivateRouteAdmin>
      <PrivateRouteAdmin
        exact={true}
        path="/admin/mainCategories"
      >
        <Unknown />
      </PrivateRouteAdmin>
      <PrivateRouteAdmin
        exact={true}
        path="/admin/subCategories"
      >
        <Unknown />
      </PrivateRouteAdmin>
      <PrivateRouteAdmin
        exact={true}
        path="/admin/updatePassword"
      >
        <Unknown />
      </PrivateRouteAdmin>
      <PrivateRouteStudent
        exact={true}
        path="/student"
      >
        <StudentPage />
      </PrivateRouteStudent>
      <PrivateRouteStudent
        exact={true}
        path="/student/updatePassword"
      >
        <UpdatePasswordPage />
      </PrivateRouteStudent>
      <Route
        component={[Function]}
        path="/teacher/:name"
      />
      <PrivateRouteTeacher
        exact={true}
        path="/my-courses"
      >
        <Unknown />
      </PrivateRouteTeacher>
      <PrivateRouteTeacher
        path="/settings"
      >
        <Unknown />
      </PrivateRouteTeacher>
      <PrivateRouteTeacher
        path="/courses/:courseId/settings"
      >
        <Unknown />
      </PrivateRouteTeacher>
      <PrivateRouteTeacher
        path="/new"
      >
        <Unknown />
      </PrivateRouteTeacher>
      <Route
        component={[Function]}
      />
    </Switch>
  </Context.Provider>
  <Memo(GlobalStyleComponent) />
</BrowserRouter>
`;
